js-api-spec/node-package-importer.node.test.ts

```diff
@@ -0,0 +1,580 @@
+// Copyright 2023 Google Inc. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import {
+  compile,
+  compileAsync,
+  compileString,
+  compileStringAsync,
+  render,
+  renderSync,
+  nodePackageImporter,
+  NodePackageImporter,
+  LegacyException,
+  LegacyResult,
+} from 'sass';
+
+import {sandbox} from './sandbox';
+
+const testPackageImporter = ({
+  input,
+  output,
+  files,
+}: {
+  input: string;
+  output: string;
+  files: {[path: string]: string};
+}) =>
+  sandbox(dir => {
+    dir.write(files);
+    dir.chdir(() => {
+      const result = compileString(input, {
+        importers: [nodePackageImporter],
+      });
+      expect(result.css).toEqualIgnoringWhitespace(output);
+    });
+  });
+
+describe('Node Package Importer', () => {
+  describe('resolves conditional exports', () => {
+    ['sass', 'style', 'default'].forEach(key => {
+      it(`${key} for root `, () =>
+        testPackageImporter({
+          input: '@use "pkg:foo";',
+          output: 'a {b: c;}',
+          files: {
+            'node_modules/foo/src/sass/_styles.scss': 'a {b: c}',
+            'node_modules/foo/package.json': JSON.stringify({
+              exports: {
+                '.': {
+                  [key]: './src/sass/_styles.scss',
+                },
+              },
+            }),
+          },
+        }));
+
+      it(`${key} for root without '.'`, () =>
+        testPackageImporter({
+          input: '@use "pkg:foo";',
+          output: 'a {b: c;}',
+          files: {
+            'node_modules/foo/src/sass/_styles.scss': 'a {b: c}',
+            'node_modules/foo/package.json': JSON.stringify({
+              exports: {
+                [key]: './src/sass/_styles.scss',
+              },
+            }),
+          },
+        }));
+
+      it(`${key} with subpath`, () =>
+        testPackageImporter({
+          input: '@use "pkg:foo/other";',
+          output: 'a {b: c;}',
+          files: {
+            'node_modules/foo/src/sass/_other.scss': 'a {b: c}',
```

Comment:
For tests with only one stylesheet, it probably makes sense to just name it "styles"